mutt/buffer.c

```diff
@@ -468,3 +468,20 @@ void mutt_buffer_seek(struct Buffer *buf, size_t offset)
   if (buf)
     buf->dptr = buf->data + offset;
 }
+
+
+/**
+ * mutt_buffer_as_rostr - Convert a buffer to a const char * "string" 
+ * @param buf Buffer to that is to be converted
+ *
+ * This method exposes Buffer's underlying data field
+ */
+const char * const mutt_buffer_as_rostr (const struct Buffer *buf) 
```

Comment:
A few comments:

1. you probably want `const char *` in the return type, as returning a variable that cannot be reassigned to makes little sense
2. I don't particularly like the name: how about `mutt_buffer_string`?
3. I don't think you get much chance for inlining, unless you've got some clever link-time optimization in place: how about defining this as inline in the header file?

Reply:
1. fixed
2. fixed
3. fixed
4. I am not sure what NONULL is. Can you explain? Especially in the context of C99 which our compile flags seem to target explicitly. ( I suspect this is some kind of gcc-only magic? )

Reply:
> I am not sure what NONULL is.

It's this little macro of ours: https://github.com/neomutt/neomutt/blob/master/mutt/string2.h#L37

Reply:
However, I now realize that mutt_b2s did the same thing, so yeah, I think it's fine to fallback to an empty string, unless we can prove that callers don't use the feture.